code/game/machinery/dna_infuser/dna_infuser.dm

```diff
@@ -138,6 +138,36 @@
 	check_tier_progression(target)
 	return TRUE
 
+/obj/machinery/dna_infuser/proc/infuse_limb(mob/living/carbon/human/target)
+	if(!ishuman(target))
+		return FALSE
+	var/obj/item/bodypart/new_limb = pick_limb(target)
+	if(!new_limb)
+		return FALSE
+	new_limb = new new_limb()
+	target.del_and_replace_bodypart(new_limb, special = TRUE)
+	new_limb.variable_color = target.hair_color
+	new_limb.update_limb(is_creating = TRUE)
+	target.updateappearance(mutcolor_update = TRUE)
+	check_tier_progression(target)
+	if(!pick_limb(target) && infusing_into.all_limbs_mutant) // no more viable limbs
+		ADD_TRAIT(target, TRAIT_MUTANT, BODYPART_TRAIT)
+	return TRUE
+
+/obj/machinery/dna_infuser/proc/pick_limb(mob/living/carbon/human/target)
+	if(!infusing_into)
+		return FALSE
```

Comment:
Returning `obj | FALSE` is a weird pattern, these should probably just be `return null` and then we can `isnull` in the proc above